src/electron/routing_service.ts

```diff
@@ -154,6 +156,22 @@ export class RoutingDaemon {
     }
   }
 
+  // Parses JSON `data` as a `RoutingServiceResponse`. Logs the error and returns undefined on
+  // failure.
+  private parseRoutingServiceResponse(data: Buffer) {
+    if (!data) {
+      console.error('received empty response from routing service');
+      return undefined;
+    }
+    let response: RoutingServiceResponse|undefined = undefined;
+    try {
+      response = JSON.parse(data.toString());
+    } catch (error) {
+      console.error(`failed to parse routing service response: ${data.toString()}`);
```

Comment:
we want to return undefined here right?

Reply:
We are, because response will be undefined.